src/System.Private.CoreLib/shared/System/Number.NumberBuffer.cs

```diff
@@ -19,22 +19,20 @@ internal static partial class Number
         [StructLayout(LayoutKind.Sequential, Pack = 1)]
         internal unsafe ref struct NumberBuffer
         {
-            public int precision;                       //  0
-            public int scale;                           //  4
-            private int _sign;                          //  8
-            private NumberBufferKind _kind;             // 12
-            private char* _allDigits;                   // 16
-            private DigitsAndNullTerminator _digits;    // 20 or 24
-
-            public bool sign { get => _sign != 0; set => _sign = value ? 1 : 0; }
-            public char* digits => (char*)Unsafe.AsPointer(ref _digits);
-            public NumberBufferKind kind { get => _kind; set => _kind = value; }
-
-            [StructLayout(LayoutKind.Sequential, Size = (NumberMaxDigits + 1) * sizeof(char))]
-            private struct DigitsAndNullTerminator { }
+            public int precision;
+            public int scale;
+            public fixed char digits[NumberMaxDigits + 1];
+            public bool sign;
```

Comment:
Why move these after digits?

Reply:
It is just ordering these by their natural alignment alignment:
* precision/scale come first because they each have 4-byte natural alignment (8-bytes total size)
* digits comes next because it is 2-byte natural alignment (102-bytes total size)
* sign and kind come last because they are 1-byte natural alignment (2-bytes total size)

Reply:
Fair enough, I'll fix this up and keep that in mind for the future.

Reply:
Fixed.